backend/services/news/src/datasources/News.ts

```diff
@@ -50,13 +25,56 @@ export async function getUploadUrl(fileName: string | undefined): Promise<Upload
 }
 
 export default class News extends dbUtils.KnexDataSource {
+  async convertArticle(article: sql.Article, keycloakId: string | undefined): Promise<gql.Article> {
+    const {
+      published_datetime,
+      latest_edit_datetime,
+      image_url,
+      body_en,
+      header_en,
+      author_id,
+      id,
+      ...rest
+    } = article;
+
+    const [likesCount] = await this.knex('article_likes')
+      .where('article_id', id)
+      .count('article_id', { as: 'likesCount' })
+      .select('article_id')
+      .groupBy('article_id');
+
+    let isLiked = false;
+    if (keycloakId) {
+      const user = await dbUtils.unique(this.knex<sql.Keycloak>('keycloak').where({ keycloak_id: keycloakId }));
+      const liked = await dbUtils.unique(this.knex<sql.Like>('article_likes')
+        .where({ article_id: id, member_id: user?.member_id }));
+      if (liked) isLiked = true;
+    }
```

Comment:
This can be written as two separate functions (haven't tested them):

```typescript
private async numberOfLikes(articleIds: UUID[]): { [key: string]: number } {
  return Object.fromEntries((await this.knex<Like>('article_likes')
    .select('article_id')
    .count({count: '*'})
    .whereIn('article_id', articleIds)
    .groupBy('article_id'))
    .map((r) => [r.article_id, r.count]))
```

```typescript
private async likedByCurrentUser(articleIds: UUID[], keycloakId: string): { [key: string]: boolean } {
  return Object.fromEntries((await this.knex<Like>('article_likes')
    .select('article_id')
    .join('keycloak', 'keycloak.member_id', '=', 'article_likes.member_id')
    .where({ keycloak_id: keycloakId })
    .whereIn('article_id', articleIds))
    .map((r) => [r.article_id, true])
```

These should be called in each function requiring the data, then passed as parameters to convertArticle making it look like this:

```typescript
export function convertArticle(article: sql.Article, numberOfLikes: number, likedByCurrentUser: boolean) {
  ...
}

...
  return {
    articles: articles.map((a) => convertArticle(a, numberOfLikes[a.id] ?? 0, likedByCurrentUser[a.id])
...
```